perceval/backends/mediawiki.py

```diff
@@ -0,0 +1,475 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import datetime
+import json
+import logging
+import os.path
+import urllib
+
+import dateutil
+import requests
+
+from ..backend import Backend, BackendCommand, metadata
+from ..cache import Cache
+from ..errors import BackendError, CacheError
+from ..utils import (DEFAULT_DATETIME,
+                     datetime_to_utc,
+                     str_to_datetime,
+                     urljoin)
+
+
+logger = logging.getLogger(__name__)
+
+MAX_RECENT_DAYS = 30  # max number of days included in MediaWiki recent changes
+
+
+class MediaWiki(Backend):
+    """MediaWiki backend for Perceval.
+
+    This class retrieves the wiki pages and edits from a
+    MediaWiki site. To initialize this class the
+    URL must be provided.
+
+    :param url: MediaWiki url
+    :param cache: cache object to store raw data
+    :param origin: identifier of the repository; when `None` or an
+        empty string are given, it will be set to `url` value
+    """
+    version = '0.1.0'
+
+    def __init__(self, url, cache=None, origin=None):
+        origin = origin if origin else url
+
+        super().__init__(origin, cache=cache)
+        self.url = url
+        self.client = MediaWikiClient(url)
+
+    def __get_max_date(self, reviews):
+        """" Get the max date in unixtime format from reviews """
+        max_ts = 0
+        for review in reviews:
+            ts = str_to_datetime(review['timestamp'])
+            ts = ts.replace(tzinfo=dateutil.tz.tzutc())
```

Comment:
Ok, analyzing the difference between what it is done in datetime_to_utc and the current approach. In this code it is missing to add to the date with the timezone before removing it (converting it to utc timezone). Changed!